spec/unit/reports_spec.rb

```diff
@@ -43,6 +43,12 @@
     Puppet::Reports.register_report(:testing) { }
   end
 
+  it "should allow a failed report to be redefined" do
+    Puppet::Reports.expects(:genmodule).returns(Module.new).twice
```

Comment:
You want to remove this expects, it blocks the error from occuring. Instead, the next line should expect an exception. If I do that I get
```
  1) Puppet::Reports  when registering report types should allow a failed report to be redefined
     Failure/Error: Puppet::Reports.register_report(:testing) { }
     Puppet::ConstantAlreadyDefined:
       Class Testing is already defined in Puppet::Reports
     # ./lib/puppet/util/classgen.rb:164:in `handleclassconst'
     # ./lib/puppet/util/classgen.rb:128:in `genthing'
     # ./lib/puppet/util/classgen.rb:60:in `genmodule'
     # ./lib/puppet/reports.rb:57:in `register_report'
     # ./spec/unit/reports_spec.rb:51:in `block (2 levels) in <top (required)>'
```
without the fix, and test passes with the fix.